src/mbgl/renderer/render_layer.hpp

```diff
@@ -81,14 +81,13 @@ class RenderLayer {
 
     friend std::string layoutKey(const RenderLayer&);
 
+    // Stores current set of tiles to be rendered for this layer.
+    std::vector<std::reference_wrapper<RenderTile>> renderTiles;
```

Comment:
`Placement` and `CrossTileSymbolIndex` both need (non-const) access to the set of render tiles -- this is how they go through updating all the opacity information.

It'd just be a band-aid, but maybe it would be better to explicitly "friend" them instead of making `renderTiles` fully public?